### PR TITLE
BLD: Check for submodules before build

### PR DESCRIPTION
**Pull request by r-devulap**
setup will look for all the submodules and complains if they aren't initialized. I wonder if we should just run the `git submodule update --init` command instead of raising an error?

### CODE REVIEW

File: numpy/core/setup.py [hunk 1]
```diff
@@ -79,11 +79,13 @@ def can_link_svml():
...
-        raise RuntimeError("Missing `SVML` submodule! Run `git submodule "
-                           "update --init` to fix this.")
-    return True
+def check_git_submodules():
+    out = os.popen("git submodule status")
+    modules = out.readlines()
+    for submodule in modules:
+        if (submodule.strip()[0] == "-"):
+            raise RuntimeError("git submodules are not initialized."
+                    "Please run `git submodule update --init` to fix this")
```
Comment (lorentzenchr) [inline numpy/core/setup.py:88]:
```suggestion
                    "Please run `git submodule update --init` to fix this.")
```

**Review by lorentzenchr — APPROVED** [review]
Sometimes, e.g. for older PRs, running `git submodule update --init` solves (build) issues.

**Comment by rgommers** [comment]
Hmm, getting Cirrus CI to actually check out the submodule turns out to be a little annoying.

**Review by rgommers — APPROVED** [review]
All green now, in it goes. Thanks @r-devulap

**Comment by r-devulap** [comment]
aarch64 CI don't need the submodules, since they only build on x86.

**Comment by rgommers** [comment]
@r-devulap neither do any architectures, but if we're venturing into multiple git submodules I think the check you implemented seems reasonable. If it becomes a noticeable slowdown, we can explore a different fix.

**Pull request merged**